File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/FellowSpectatorJoinedPacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FellowSpectatorJoinedPacket implements ServerPacket {
  private int userId;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_FELLOW_SPECTATOR_JOINED;
  }

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    writer.writeInt32(stream, userId);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/FellowSpectatorJoinedPacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.core.ServerPacket
import pe.nanamochi.banchus.io.DataWriter

class FellowSpectatorJoinedPacket(var userId: Int = 0) : ServerPacket() {
    override val type = PacketType.BANCHO_FELLOW_SPECTATOR_JOINED

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeInt32(out, userId)
    }
}
